Add compatibility documentation (#604)
kyma-project/cli: docs/internal/compatibility-template.md

# Compatibility templates

## Full compatibility

The Kyma CLI is always released in parallel with Kyma to support the latest features, which also means that older Kyma versions can no longer be supported by the Kyma CLI.
Unless otherwise stated in the documentation of a specific Kyma CLI version, the Kyma CLI will always support the Kyma version it was released for and be backwards compatible with the two previous Kyma releases.

## Partial compatibility

> The Kyma CLI is always released in parallel with Kyma to support the latest features, which also means that older Kyma versions can no longer be supported by the Kyma CLI.

Reviewer: ```suggestion

The Kyma CLI is always released in parallel with Kyma to support the latest features, which also means that older Kyma versions can no longer be supported by the Kyma CLI.
```

> This Kyma CLI version is compatible with the corresponding Kyma release and the previous release, but it is incompatible with versions before the previous ones.

Reviewer: ```suggestion
This Kyma CLI version is compatible with the corresponding Kyma release and the previous release, but it is incompatible with versions before the last but one.
```
or `before the two previous ones`. 
All versions before the current one are previous versions. If we refer to more than one, it's not clear how many previous ones we mean, so we need to specify.

> ## Incompatible

Reviewer: ```suggestion
## Incompatible

```

The Kyma CLI is always released in parallel with Kyma to support the latest features, which also means that older Kyma versions can no longer be supported by the Kyma CLI.

> This Kyma CLI version is only compatible with the corresponding Kyma release and incompatible with all previous versions.

Reviewer: ```suggestion
This Kyma CLI version is only compatible with the corresponding Kyma release and is incompatible with all previous versions.
```
